File: Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/procesado_datos.py
# -*- coding: utf-8 -*-
"""
Created on Tue May 17 12:38:06 2022

@author: Abigail
"""

#Plantilla de Pre Procesado

#Cómo importar las librerías

import numpy as np #matemáticas
import matplotlib.pyplot as plt #graficas y representación visual
import pandas as pd #carga de datos

#importar el dataset
dataset = pd.read_csv('Data.csv')
#iloc sirve para localizar los elemntos (filas y columnas) de un sataset
#por posicion i=index
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 3].values


#Dividir el data set en conjunto d eentrenamiento y conjunto de test
#size=0.2 implica que un 20% de los datos se destinaran a testing, max0.3
from sklearn.model_selection import train_test_split
X_train, X_test, y_trai, y_test = train_test_split(X, y, test_size=0.2, random_state=0)



#Escalado de variables 
#Se hace a menudo pero no siempre
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
#solo transform para que detecte la transformaci+on que se hizo en el de entrenamiento
X_test = sc_X.transform(X_test)"""

File: Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/procesado_datos.R
#Plantilla para el pre procesado de datos
#Importar el data set

dataset = read.csv('Data.csv')


#Dividir los datos en conjunto de entrenamiento 
install.packages("caTools")

library(caTools)
set.seed(123) #semilla aleatoria
#0.8 es el porcentaje que sive para entrenar (train)
split = sample.split(dataset$Purchased, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
testing_set = subset(dataset, split == FALSE)

#para escalar los datos, se puede estandarizar o normalizar 
#Estandarizar permite aglutinar valores en torno a la media, da la campana de Gaus
# Normalización, transforma la data de 0 a 1, el valor más grnade tomará el valor de 1
#y el más pequeño en 0, se escala de forma lineal



##Escalado de valores (normalización)
### normalmente se hace, pero no siempre
# training_set [, 2:3] = scale(training_set[,2:3])
# testing_set [, 2:3]= scale(testing_set[, 2:3])













